amqp/Go/amqpGet.go

```diff
@@ -0,0 +1,181 @@
+/*  Unless required by applicable law or agreed to in writing, software                 */
+/*  distributed under the License is distributed on an "AS IS" BASIS,                   */
+/*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.            */
+/*  See the License for the specific language governing permissions and                 */
+/*  limitations under the License.                                                      */
+/*                                                                                      */
+/*                                                                                      */
+/****************************************************************************************/
+/*                                                                                      */
+/*  FILE NAME:      amqpGet.go                                                          */
+/*                                                                                      */
+/*  DESCRIPTION:    A Go applicaton used to receive messages from broker-IBM MQ.        */
+/*  AMQP 1.0 in use. Go-azure library is used.                                          */
+/*  Refer : https://github.com/Azure/go-amqp                                            */
+/*  Documentation: https://pkg.go.dev/github.com/Azure/go-amqp#section-documentation    */
+/*                                                                                      */
+/*  How to Run:                                                                         */
+/*  Usage:  go run amqpGet.go [-t TopicName] [-q QueueName]                             */
+/*                                                                                      */
+/*  Ex:  go run amqpGet.go -q LOCAL.QUEUE.1                                             */
+/*                                                                                      */
+/*  Parameters:                                                                         */
+/*  -t ==> TopicName                                                                    */
+/*  -q ==> QueueName                                                                    */
+/****************************************************************************************/
+ 
+package main
+
+import (
+    "log"
+    "fmt"
+    "context"
+    "flag"
+    "encoding/json"
+    "io/ioutil"
+    "os"
+
+    "github.com/Azure/go-amqp"
+)
+
+type Env struct {
+    Username         string `json:"APP_USER"`
+    Password         string `json:"APP_PASSWORD"`
+    Host             string `json:"HOST"`
+    Port             string `json:"PORT"`
+}
+
+type MQEndpoints struct {
+    Points []Env `json:"MQ_ENDPOINTS"`
+}
+
+var (
+    queueName   string
+    topicName   string
+    valid       bool
+    destination string
+    isQueue     bool
+)
+var EnvSettings Env
+var MQ_ENDPOINTS MQEndpoints
+
+func init() {
+    //Parsing the input

```

Comment:
Break down the code into further methods / functions to make each logical unit of functionality easier to read.